docs/build-og-images.js

```diff
@@ -0,0 +1,116 @@
+const path = require('path');
+const fs = require('fs');
+const { createCanvas, loadImage, registerFont } = require('canvas');
+
+registerFont('static/font/Aeonik-Bold.otf', { family: 'Aeonik Bold' });
+
+const getMarkdownHeader = (path) => {
+  const content = fs.readFileSync(path, 'utf-8');
+  const headers = content
+    .split('\n')
+    .map((line) => line.trim())
+    .filter((line) => line.startsWith('#'))
+    .map((line, index) => ({
+      level: line.match(/#/g).length,
+      title: line.replace(/#/g, '').trim(),
+      index,
+    }))
+    .sort((a, b) => a.level - b.level || a.index - b.index);
+
+  return headers[0]?.title || '';
+};
+
+const buildOGImages = async () => {
+  const baseDocsPath = path.resolve(__dirname, 'docs');
+  const docs = await Promise.all(
+    (
+      await fs.promises.readdir(baseDocsPath)
+    ).map(async (dir) => {
+      const files = await fs.promises.readdir(path.resolve(baseDocsPath, dir));
+      return {
+        dir,
+        files: files.filter(
+          (file) => file.endsWith('.md') || file.endsWith('.mdx')
+        ),
+      };
+    })
+  );
+
+  const targetDocs = path.resolve(__dirname, 'build/docs/og');
```

Comment:
```suggestion
  const targetDocs = path.resolve(__dirname, 'build/img/og');
```
i would keep the images in img folder